app/persona/api/offer/Offer.scala

```diff
@@ -1,16 +1,26 @@
 package persona.api.offer
 
+import java.util.UUID
+
+import org.joda.time.DateTime
 import persona.api.authentication.User
+import persona.api.offer.offerImpl.InvalidOfferException
 import play.api.libs.concurrent.Execution.Implicits._
 
 import scala.concurrent.Future
 
-class Offer (val id: Long,
+case class Offer (val id: UUID,          // This is a timeUUID that has the exact time it was created
+             creationDay: DateTime, // TODO: may have to change to string type/check to match Cassandra
              description: String,
-             creationTime: Long,
-             expirationTime: Long,
+             expirationTime: DateTime,
              currentParticipants: Int,
-             maxParticipants: Int) {
+             maxParticipants: Int,
+             value: Double,
+             criteria: Map[String,String]) {
+
+  if(criteria.isEmpty) {
```

Comment:
Recommend changing to require(criteria.nonEmpty).  That way we don't need InvalidOfferException